### PR TITLE
Improve error catching in UI, and improve consistency across UI

### PR DESCRIPTION
**Pull request by MKuijpers**
Adds more error catching to prevent crashes
Also makes some parts of the UI more consistent, like BTC vs Satoshi

### CODE REVIEW

File: currencyii/src/main/java/nl/tudelft/trustchain/currencyii/coin/WalletManager.kt [hunk 1]
```diff
@@ -112,9 +112,13 @@ class WalletManager(
...
 
         Log.i("Coin", "Coin: starting the setup of kit.")
-        kit.setBlockingStartup(false)
-            .startAsync()
-            .awaitRunning()
+        try {
+            kit.setBlockingStartup(false)
+                .startAsync()
+                .awaitRunning()
+        } catch (t: Throwable) {
```
Comment (PJvanderLaan) [inline currencyii/src/main/java/nl/tudelft/trustchain/currencyii/coin/WalletManager.kt]: Why do we want to catch the exception here?

File: currencyii/src/main/java/nl/tudelft/trustchain/currencyii/ui/bitcoin/CreateSWFragment.kt [hunk 2]
```diff
@@ -79,15 +83,18 @@ class CreateSWFragment() : BaseFragment(R.layout.fragment_create_sw) {
...
+        }
     }
 
     private fun validateCreationInput(): Boolean {
-        val entranceFee = entrance_fee_tf.text.toString().toDoubleOrNull()
+        val entranceFee = entrance_fee_tf.text.toString().toLongOrNull()
         val votingThreshold = voting_threshold_tf.text.toString().toIntOrNull()
         return entranceFee != null &&
-            entranceFee > 0 &&
+            entranceFee >= 5000 &&
```
Comment (PJvanderLaan) [inline currencyii/src/main/java/nl/tudelft/trustchain/currencyii/ui/bitcoin/CreateSWFragment.kt]: should be in a (static?) constant

File: currencyii/src/main/java/nl/tudelft/trustchain/currencyii/ui/bitcoin/SharedWalletTransaction.kt [hunk 3]
```diff
@@ -62,7 +62,7 @@ class SharedWalletTransaction : BaseFragment(R.layout.fragment_shared_wallet_tra
     }
 
     private fun transferFundsClicked() {
-        if (!validateCreationInput()) {
+        if (!validateTransferInput()) {
             activity?.runOnUiThread {
                 alert_view.text = "Failed: Bitcoin PK should be a string, minimal satoshi amount: 5000"
```
Comment (PJvanderLaan) [inline currencyii/src/main/java/nl/tudelft/trustchain/currencyii/ui/bitcoin/SharedWalletTransaction.kt]: Use the (not yet present) min satoshi constant here to display the correct min. satoshi amount

File: currencyii/src/main/java/nl/tudelft/trustchain/currencyii/ui/bitcoin/SharedWalletTransaction.kt [hunk 4]
```diff
@@ -163,7 +171,7 @@ class SharedWalletTransaction : BaseFragment(R.layout.fragment_shared_wallet_tra
         }
     }
 
-    private fun validateCreationInput(): Boolean {
+    private fun validateTransferInput(): Boolean {
         val bitcoinPublicKey = input_bitcoin_public_key.text.toString()
         val satoshiTransferAmount = input_satoshi_amount.text.toString().toLong()
         return bitcoinPublicKey != "" && satoshiTransferAmount >= 5000 && blockHash != null
```
Comment (PJvanderLaan) [inline currencyii/src/main/java/nl/tudelft/trustchain/currencyii/ui/bitcoin/SharedWalletTransaction.kt]: also here, '5000'

**Review by PJvanderLaan — APPROVED** [review]
Looking good, some small comments

File: currencyii/src/main/java/nl/tudelft/trustchain/currencyii/coin/WalletManager.kt (hunk 1 above)
Comment (MKuijpers) [reply currencyii/src/main/java/nl/tudelft/trustchain/currencyii/coin/WalletManager.kt]: Before the stability fixes, I used this to see what caused the crashes. I can remove the try-catch, need to make sure they are caught somewhere else. I will see if I can remove it and catch it where they are called from the UI, so we can show the user that something went wrong.

File: currencyii/src/main/java/nl/tudelft/trustchain/currencyii/ui/bitcoin/CreateSWFragment.kt (hunk 2 above)
Comment (MKuijpers) [reply currencyii/src/main/java/nl/tudelft/trustchain/currencyii/ui/bitcoin/CreateSWFragment.kt]: Yup, moving it to a constant in SWUtil

File: currencyii/src/main/java/nl/tudelft/trustchain/currencyii/ui/bitcoin/SharedWalletTransaction.kt (hunk 3 above)
Comment (MKuijpers) [reply currencyii/src/main/java/nl/tudelft/trustchain/currencyii/ui/bitcoin/SharedWalletTransaction.kt]: Will do

File: currencyii/src/main/java/nl/tudelft/trustchain/currencyii/ui/bitcoin/SharedWalletTransaction.kt (hunk 4 above)
Comment (MKuijpers) [reply currencyii/src/main/java/nl/tudelft/trustchain/currencyii/ui/bitcoin/SharedWalletTransaction.kt]: See comment above.

File: currencyii/src/main/java/nl/tudelft/trustchain/currencyii/ui/bitcoin/CreateSWFragment.kt (hunk 2 above)
Comment (MKuijpers) [reply currencyii/src/main/java/nl/tudelft/trustchain/currencyii/ui/bitcoin/CreateSWFragment.kt]: Fixed

File: currencyii/src/main/java/nl/tudelft/trustchain/currencyii/ui/bitcoin/SharedWalletTransaction.kt (hunk 3 above)
Comment (MKuijpers) [reply currencyii/src/main/java/nl/tudelft/trustchain/currencyii/ui/bitcoin/SharedWalletTransaction.kt]: Fixed

File: currencyii/src/main/java/nl/tudelft/trustchain/currencyii/ui/bitcoin/SharedWalletTransaction.kt (hunk 4 above)
Comment (MKuijpers) [reply currencyii/src/main/java/nl/tudelft/trustchain/currencyii/ui/bitcoin/SharedWalletTransaction.kt]: Fixed

File: currencyii/src/main/java/nl/tudelft/trustchain/currencyii/coin/WalletManager.kt (hunk 1 above)
Comment (PJvanderLaan) [reply currencyii/src/main/java/nl/tudelft/trustchain/currencyii/coin/WalletManager.kt]: Yes better to not catch the error here

Comment (MKuijpers) [reply currencyii/src/main/java/nl/tudelft/trustchain/currencyii/coin/WalletManager.kt]: Fixed: https://github.com/InvictusRMC/trustchain-superapp/pull/14/commits/f47db8d093de2d3a8e5e2532b68a870dd47f96dd

**Pull request merged**